### PR TITLE
spod: Make /etc/selinux.d an emptyDir instead of hostPath

### PR DESCRIPTION
**Pull request by jhrozek**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Since the directory only contains policy definitions, not the actual
policy result it is not needed to be a hostPath. The actual policy is
persisted in /etc/selinux (without the .d).

The original idea behind having a hostPath was to run selinuxd on the
nodes directly to lock down even e.g. the kubelet. But that will not
happen, at least not in the short term, so let's reduce the number of
our hostPath mounts a little.

#### Which issue(s) this PR fixes:
None

#### Does this PR have test?
N/A tested by existing tests

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?

```release-note
The directory /etc/selinux.d used to be mounted on the hosts in previous SPO versions.
This is no longer the case, the directory was converted to an emptyDir instead,
reducing the number of required host mounts.

```

### CODE REVIEW

**Review by JAORMX — COMMENTED** [review]
lgtm

I'll merge it once it passes CI

**Review by saschagrunert — APPROVED** [review]

**Comment by JAORMX** [comment]
/hold

debugging this a little before we merge

**Pull request merged**
